lib/leaf.go

```diff
@@ -47,15 +54,31 @@ func NewNamedLeaf(name string, structPtr interface{}) *Leaf {
 		structureType:    getStructureType(structPtr),
 		structureValue:   getStructureValue(structPtr),
 		structureElement: getStructureElement(structPtr),
+		structurePointer: getStructurePointer(structPtr),
 		name:             name,
+		aliases:          make(map[string]struct{}),
 	}
 
 	leaf.initializeDependencies()
 	leaf.initializePostConstruct()
 	leaf.initializePreDestroy()
+
+	leaf.AddAlias(leaf.name)
+
 	return leaf
 }
 
+// AddAlias adds passed name to aliases
+func (l *Leaf) AddAlias(name string) {
```

Comment:
Does this need to be public? I'm not sure we want to expose this functionality, all leaf additions should go through a tree.